src/grt/src/RepairAntennas.cpp

```diff
@@ -1090,431 +954,521 @@ void RepairAntennas::addJumperVertical(const int& seg_id,
+    int adj_layer_level = adj_it.first->getRoutingLevel();
+    if (adj_layer_level != layer_level) {
+      int adj_seg_id = segment_graph[adj_it.first][adj_it.second].seg_id;
+      if (is_horizontal) {
+        via_pos.insert(route[adj_seg_id].init_x);
+      } else {
+        via_pos.insert(route[adj_seg_id].init_y);
+      }
+    }
+  }
+}
 
-  // Get violation info
-  ViolationIdToSegmentIds segment_with_violations;
-  odb::dbTech* tech = db_->getTech();
+int RepairAntennas::getBestPosition(const std::vector<int>& position_cand,
+                                    const bool& is_horizontal,
+                                    const odb::Point& parent_pos)
+{
+  int jumper_position = -1;
+  // Find best position of the position candidates
+  int min_dist = -1, dist;
+  for (const int& pos : position_cand) {
+    // Get distance
+    if (is_horizontal) {
+      dist = std::abs((pos + tile_size_) - parent_pos.x());
+    } else {
+      dist = std::abs((pos + tile_size_) - parent_pos.y());
+    }
+    if (min_dist == -1 || dist < min_dist) {
+      min_dist = dist;
+      jumper_position = pos;
+    }
+  }
+  return jumper_position;
+}
 
-  // Init DSU
-  std::vector<int> dsu_parent(added_seg_count);
-  std::vector<int> dsu_size(added_seg_count);
-  for (int i = 0; i < added_seg_count; i++) {
-    dsu_size[i] = 1;
-    dsu_parent[i] = i;
+bool RepairAntennas::findPosToJumper(const GRoute& route,
+                                     LayerToSegmentNodeVector& segment_graph,
+                                     const SegmentNode& seg_node,
+                                     const odb::Point& parent_pos,
+                                     int& jumper_position)
+{
+  jumper_position = -1;
+  const GSegment& seg = route[seg_node.seg_id];
+  // Ignore small segments
+  if (seg.length() < smaller_seg_size_) {
+    return false;
   }
-  boost::disjoint_sets<int*, int*> dsu(&dsu_size[0], &dsu_parent[0]);
-
-  // Run DSU
-  int min_layer = 1;
-  odb::dbTechLayer* layer_iter = tech->findRoutingLayer(min_layer);
-  for (; layer_iter; layer_iter = layer_iter->getUpperLayer()) {
-    // iterate each node of this layer to union set
-    for (auto& seg_it : segment_by_layer[layer_iter]) {
-      const int& id_u = seg_it.id;
-      // get neighbors and union
-      for (const auto& adj_it : seg_it.adjs) {
-        const int& id_v = segment_by_layer[adj_it.first][adj_it.second].id;
-        // if they are on different sets then union
-        if (dsu.find_set(id_u) != dsu.find_set(id_v)) {
-          dsu.union_set(id_u, id_v);
-        }
+  // Get init and final position of segment
+  const int seg_init_x = seg.init_x;
+  const int seg_init_y = seg.init_y;
+  const int seg_final_x = seg.final_x;
+  const int seg_final_y = seg.final_y;
+  const int step = tile_size_;
+
+  int pos_x = seg_init_x;
+  int pos_y = seg_init_y;
+  int last_block_x = pos_x;
+  int last_block_y = pos_y;
+  bool has_available_resources, is_via;
```

Comment:
Avoid multiple declarations in a single line. Tho it saves line count, I find it quite easy to miss one of the declarations.